NDTensors/src/dense/tensoralgebra/contract.jl

```diff
@@ -175,14 +175,18 @@ function _contract_scalar_perm!(
     if iszero(α)
       fill!(Rᵃ, 0)
     else
-      @strided Rᵃ .= α .* permutedims(Tᵃ, perm)
+      Rᵃ = permutedims!!(
+        leaf_parenttype(Rᵃ), Rᵃ, leaf_parenttype(Tᵃ), Tᵃ, perm, (r, t) -> *(α, t)
+      )
     end
   elseif isone(β)
     if iszero(α)
       # Rᵃ .= Rᵃ
       # No-op
     else
-      @strided Rᵃ .= α .* permutedims(Tᵃ, perm) .+ Rᵃ
+      Rᵃ = permutedims!!(
+        leaf_parenttype(Rᵃ), Rᵃ, leaf_parenttype(Tᵃ), Tᵃ, perm, (r, t) -> r + α * t
```

Comment:
To simplify this kind of code, can't we write it as:
```julia
      Rᵃ = permutedims!!(Rᵃ, Tᵃ, perm, (r, t) -> r + α * t)
```
and then define a function:
```julia
permutedims!!(R, T, perm, f) = permutedims!!(leaf_parenttype(R), R, leaf_parenttype(T), T, perm, f)
```
which does the trait dispatch. That would also make it a bit easier to switch to `SimpleTraits.jl` if we want.